### ISSUE TITLE
MPS nn.BatchNorm2d gives different result from CPU

### ISSUE THREAD

**Issue by m-reuter** [pytorch/pytorch#84962]
### 🐛 Describe the bug

Running batchnorm2d gives different results for me depending on mps or cpu device: 

```python
import torch
import torch.nn
bn = torch.nn.BatchNorm2d(3,affine=False)
x = torch.randn(20, 3, 35, 45)
y = bn(x)
bn2 = bn.to("mps")
x2 = x.to("mps")
y2 = bn2(x2)
y2 = y2.to("cpu")
print (torch.max(torch.abs(y-y2)))
print(torch.max(y),torch.max(y2))
```

Gives:
```
tensor(0.5964)
tensor(4.1934) tensor(3.6132)
```
(due to randn your numbers will differ from these, but more importantly y and y2 should be the same. )
Or do I need to cast something here?

Thanks for your help. This was tested on the current nightly build. 

### Versions

Collecting environment information...
PyTorch version: 1.13.0.dev20220913
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.4 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: Could not collect
Libc version: N/A

Python version: 3.8.9 (default, Apr 13 2022, 08:48:06)  [Clang 13.1.6 (clang-1316.0.21.2.5)] (64-bit runtime)
Python platform: macOS-12.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.2
[pip3] torch==1.13.0.dev20220913
[pip3] torchio==0.18.83
[pip3] torchvision==0.14.0.dev20220913
[conda] Could not collect

cc @kulinseth @albanD

**Comment by kulinseth** [comment]
@DenisVieriu97 , please take a look.

**Comment by princeminz** [comment]
Unable to replicate this issue on my m1 mac on the latest nightly.

Ran the following code:
```
import torch
import torch.nn
bn = torch.nn.BatchNorm2d(3,affine=False)
x = torch.randn(20, 3, 35, 45)
y = bn(x)
bn2 = bn.to("mps")
x2 = x.to("mps")
y2 = bn2(x2)
y2 = y2.to("cpu")
print (torch.max(torch.abs(y-y2)))
print(torch.max(y),torch.max(y2))
```
which gave the output:
```
tensor(4.7684e-07)
tensor(4.2712) tensor(4.2712)
```
### Versions
```
Collecting environment information...
PyTorch version: 1.13.0.dev20220914
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.5.1 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:41:22) [Clang 13.0.1 ] (64-bit runtime)
Python platform: macOS-12.5.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.2
[pip3] torch==1.13.0.dev20220914
[pip3] torchaudio==0.13.0.dev20220914
[pip3] torchvision==0.14.0.dev20220914
[conda] numpy                     1.23.2          py310h127c7cf_0    fastchan
[conda] pytorch                   1.13.0.dev20220914        py3.10_0    pytorch-nightly
[conda] torchaudio                0.13.0.dev20220914       py310_cpu    pytorch-nightly
[conda] torchvision               0.14.0.dev20220914       py310_cpu    pytorch-nightly
```

**Comment by m-reuter** [comment]
I am on 
Model Identifier:	MacBookPro18,2
  Chip:	Apple M1 Max
  Total Number of Cores:	10 (8 performance and 2 efficiency)
  Memory:	32 GB
  System Firmware Version:	7459.121.3
  OS Loader Version:	7459.121.3

with 32 GPU cores.

**Comment by m-reuter** [comment]
I just updated the OS to Monterey 12.6
python --version is 
Python 3.9.6

and:
```
pip freeze
certifi==2022.9.14
charset-normalizer==2.1.1
idna==3.4
numpy==1.23.3
Pillow==9.2.0
requests==2.28.1
torch==1.13.0.dev20220913
torchvision==0.14.0.dev20220914
typing_extensions==4.3.0
urllib3==1.26.12
```

Same large difference as in my original post. Maybe it is the M1-Max?

**Comment by ekolve** [comment]
I also observe a difference in the `y` and `y2` values:

```python
tensor(0.2283)
tensor(4.3101) tensor(4.0818)
```

  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro18,1
  Chip:	Apple M1 Pro
  Total Number of Cores:	10 (8 performance and 2 efficiency)
  Memory:	32 GB
  System Firmware Version:	7459.141.1
  OS Loader Version:	7459.141.1
  Activation Lock Status:	Disabled

Python 3.10.6
torch==1.13.0a0+git2386cd2

**Comment by kulinseth** [comment]
@m-reuter , thanks for the issue. We were able to reproduce this issue locally and it was indeed a bug for BatchNorm implementation (for these specific sizes) in MPS which is fixed in MacOS Ventura.

**Comment by m-reuter** [comment]
Thanks @kulinseth . I updated to Ventura and it works there.
